spec/models/profile_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe Profile do
+  let(:user) { create(:user, :signed_up) }
+  let(:profile) do
+    Profile.create(
+      user_id: user.id
+    )
+  end
+
+  subject { profile }
+
+  it { is_expected.to belong_to(:user) }
+
+  describe 'allows only one active Profile per user' do
+    it 'prevents create! via ActiveRecord uniqueness validation' do
+      profile.active = true
+      profile.save!
+      expect do
+        Profile.create!(user_id: user.id, active: true)
+      end.to raise_error(ActiveRecord::RecordInvalid)
+    end
+
+    it 'prevents save! via psql unique partial index' do
+      profile.active = true
```

Comment:
Sorry, I was unclear. It's not `save` vs `save!` so much here as the `validate: false` piece. I.e. I want to confirm if psql is going to raise an exception, which it will regardless of `save` or `save!`. But I can't just check the return value, or the whole test itself will abort because it's a runtime exception.